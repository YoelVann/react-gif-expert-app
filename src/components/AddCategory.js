import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState("");
    const handleInputChange = (e) =>{
        
        setinputValue(e.target.value)
    }

    const handleSubmmit = (e) =>{
        e.preventDefault();
        if( inputValue.trim().length>2){
            setCategories(cats => [inputValue,...cats]);
            setinputValue('');
        }
    }
  return (
      <form onSubmit={ handleSubmmit }>
        <h1>{inputValue}</h1>
        <input 
            type="text"
            placeholder="Escribe una categorÃ­a Ej: 'Black Clover' y presiona Enter"
            value={inputValue}
            onChange={handleInputChange}
        />
      </form>
  )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
