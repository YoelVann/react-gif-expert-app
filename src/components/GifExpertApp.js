import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const [categories, setCategories] = useState(['Naruto']);
    const [categories, setCategories] = useState(defaultCategories);
    
    // const handleAdd = () =>{
    //     setCategories([...categories, 'Hunter X Hunter']);
    // }

    return ( 
        <>
        <h2>Gif Expert App</h2>
        <AddCategory setCategories={setCategories}/>
        <hr></hr>
        <ol>
            {
                categories.map( category => 
                    <GifGrid 
                        key={category}
                        category={category}
                    />
                )    
            }
        </ol>
    
        </>
    );
};

export default GifExpertApp;
