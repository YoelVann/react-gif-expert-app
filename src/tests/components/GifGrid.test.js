import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GifGridItem', () => { 

    const category = "Black Clover"

    test('Debe de mostrar el componente', () => { 
        
       useFetchGifs.mockReturnValue({
        data:[],
        loading: true
       });
        const wrapper = shallow( <GifGrid  category={category}/>);
        expect( wrapper ).toMatchSnapshot();
    });
   
    
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => { 
        
        const images = [{
            id: 'ABC',
            url: 'https:localhost/hola/hola.jpg',
            title: 'title'
        }];

        useFetchGifs.mockReturnValue({
            data: images,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid  category={category}/>);
       
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(images.length)
    });
 });
