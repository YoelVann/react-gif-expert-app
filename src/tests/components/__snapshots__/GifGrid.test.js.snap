// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de mostrar el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Black Clover
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en GifGridItem debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Black Clover
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="title"
      url="https:localhost/hola/hola.jpg"
    />
  </div>
</Fragment>
`;
